include/pmacc/particles/operations/splitIntoListOfFrames.kernel

```diff
@@ -47,192 +56,402 @@ namespace kernel
+} // namespace kernel
 
-    /* counter is used to apply for work, count used frames and count loaded particles
-     * [0] -> offset for loading particles
-     * [1] -> number of loaded particles
-     * [2] -> number of used frames
+    /** Copy particles from big frame to PMacc frame structure
+     *  (Opposite to ConcatListOfFrames)
+     *
+     * - convert a user-defined domainCellIdx to localCellIdx
+     * - processed particles per block <= number of cells per superCell
      *
-     * all values are zero after initialization
+     * @tparam T_LogLvl type of the loc level for debuging output
+     * @tparam T_DestSpecies pmacc:ParticlesBase, type of the destination species
+     * @tparam T_SrcFrame pmacc:ParticlesBox, type of the source particle frame
+     * @tparam T_Space pmacc::DataSpace, type for indicies and offsets within the domain
+     * @tparam T_Identifier Identifier, type of the identifier for the total domain offset
+     * @tparam T_CellDescription pmacc::MappingDescription, type of the domain description
+     *
+     * @param destSpecies particle species instance whose deviceBuffer is written
+     * @param srcFrame device frame with particles (is used as source)
+     * @param numParticles number of particles in srcFrame
+     * @param chunkSize number of particles to process in one kernel call
+     * @param localDomainCellOffset offset in cells to user-defined domain (@see wiki PIConGPU domain definitions)
+     * @param domainCellIdxIdentifier the identifier for the particle domain cellIdx
+     *                                that is calculated back to the local domain
+     *                                with respect to localDomainCellOffset
+     * @param cellDesc supercell domain description
+     * @param logLvl Log level used for information logging
      */
-    GridBuffer<uint32_t, DIM1> counterBuffer(DataSpace<DIM1>(3));
+    template<
+        typename T_LogLvl,
+        typename T_DestSpecies,
+        typename T_SrcFrame,
+        typename T_Space,
+        typename T_Identifier,
+        typename T_CellDescription
+    >
+    HINLINE void splitIntoListOfFrames(
+        T_DestSpecies & destSpecies,
+        T_SrcFrame srcFrame,
+        uint32_t numParticles,
+        uint32_t const chunkSize,
+        T_Space const & localDomainCellOffset,
+        T_Identifier const domainCellIdxIdentifier,
+        T_CellDescription const & cellDesc,
+        T_LogLvl const & logLvl = T_LogLvl( )
+    )
+    {
+        using SuperCellSize = typename T_CellDescription::SuperCellSize;
+        uint32_t const cellsInSuperCell = pmacc::math::CT::volume< SuperCellSize >::type::value;
+
+        /* counter is used to apply for work, count used frames and count loaded particles
+         * [0] -> offset for loading particles
+         * [1] -> number of loaded particles
+         * [2] -> number of used frames
```

Comment:
This reason is understandable but very sad. Like we are using black high performance template meta programming C++11 magic, want fixed types, but never want to write them down __anywhere__ and abstract them so far away that we feel like using Python on brace stereoids and then we pass an array of some floating point type to be compatbile with a 25 years old interface...